### DISCUSSION TITLE
BleScanner as Broadcast for tab navigation

### DISCUSSION THREAD

**Discussion by R-Emagina** [PhilipsHue/flutter_reactive_ble#436] [Q&A] [answered]
Hi,

Being quite a newbie with Flutter and and even development in general, I'll ask for your patience :)
I have used bits of the example in my own app which is using [GoRouter](https://pub.dev/packages/go_router) to navigate through tabs and pages.
An issue I've been having for the past couple of days is that, when switching from by BLE tab (basically the DevicesList from the example) to any other tab and then switching back, I was having an error : "Bad state: Stream has already been listened to.".
After a few tests and some googling, I've tried changing the BleScanner `_stateStreamController` to `StreamController.broadcast()`.
It is working, but I'm not sure why and what side effects this change might have.

Thanks for your help!

**Comment by werediver** [comment, marked as answer]
@R-Emagina There are two kinds of streams in Dart: single-subscription streams (default) and broadcast streams. The former kind allows only one subscription to be made; not one at a time, but just one, and when it's closed the stream cannot be used anymore. The latter allows multiple subscriptions to be made.

But the real issue (likely) is that you're not "caching" you subscription. If you save it for future use, you don't need another.

**Reply by R-Emagina** [reply]
Ok thank you for this answer. How do I go about caching the subscription?
Do you have any articles/knowledge bases to recommend on this subject?

Thanks again!
